Observer/QuoteSubmitSuccessObserver.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * @category    ClassyLlama
+ * @copyright   Copyright (c) 2018 Classy Llama Studios, LLC
+ */
+
+namespace ClassyLlama\AvaTax\Observer;
+
+use ClassyLlama\AvaTax\Model\Logger\AvaTaxLogger;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use ClassyLlama\AvaTax\Api\Data\AssociatedTaxableInterfaceFactory as AssociatedTaxableFactory;
+use ClassyLlama\AvaTax\Api\Data\AssociatedTaxableInterface;
+use ClassyLlama\AvaTax\Model\Tax\AssociatedTaxableRepository;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Quote\Model\Quote;
+use Magento\Sales\Model\Order;
+
+class QuoteSubmitSuccessObserver implements ObserverInterface
+{
+    /**
+     * @var \ClassyLlama\AvaTax\Helper\AssociatedTaxable
+     */
+    protected $associatedTaxableHelper;
+
+    /**
+     * @var AssociatedTaxableFactory
+     */
+    protected $associatedTaxableFactory;
+
+    /**
+     * @var AssociatedTaxableRepository
+     */
+    protected $associatedTaxableRepository;
+
+    /**
+     * @var AvaTaxLogger
+     */
+    protected $avataxLogger;
+
+    /**
+     * QuoteSubmitSuccessObserver constructor.
+     *
+     * @param AssociatedTaxableFactory    $associatedTaxableInterfaceFactory
+     * @param AssociatedTaxableRepository $associatedTaxableRepository
+     * @param AvaTaxLogger                $logger
+     */
+    public function __construct(
+        AssociatedTaxableFactory $associatedTaxableInterfaceFactory,
+        AssociatedTaxableRepository $associatedTaxableRepository,
+        AvaTaxLogger $logger,
+        \ClassyLlama\AvaTax\Helper\AssociatedTaxable $associatedTaxableHelper
+    ) {
+        $this->associatedTaxableFactory = $associatedTaxableInterfaceFactory;
+        $this->associatedTaxableRepository = $associatedTaxableRepository;
+        $this->avataxLogger = $logger;
+        $this->associatedTaxableHelper = $associatedTaxableHelper;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var Quote $quote */
+        $quote = $observer->getQuote();
```

Comment:
Wrap all of this in a try/catch, where the type you catch is `\Throwable`. Then log the error to the AvaTax logs with a high severity level (may be "critical"?).